### PR TITLE
refactor: Reduce variable usage in `backend-service` and `core-infra` examples

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Reduce variable usage in `backend-service` and `core-infra` examples
- `code*` IAM roles now default to using name prefix to ensure name collisions do not occur

## Motivation and Context
- Simplify examples to focus on the content delivered and less on the Terraform aspect

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: application-code/container-queue-proc/src/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 boto3==1.24.58
-Pillow==9.2.0
+Pillow==9.3.0
```
Comment (bryantbiggs) [inline application-code/container-queue-proc/src/requirements.txt:2]: fluffy

File: examples/backend-service/main.tf [hunk 2]
```diff
@@ -113,34 +79,27 @@ resource "aws_service_discovery_service" "sd_service" {
...
   source = "../../modules/ecs-service"
 
-  name                       = var.service_name
-  desired_count              = var.desired_count
-  ecs_cluster_id             = data.aws_ecs_cluster.core_infra.cluster_name
-  cp_strategy_base           = var.cp_strategy_base
-  cp_strategy_fg_weight      = var.cp_strategy_fg_weight
-  cp_strategy_fg_spot_weight = var.cp_strategy_fg_spot_weight
+  name           = local.name
+  desired_count  = 3
```
Comment (allamand) [inline examples/backend-service/main.tf:83]: Is it not good ta be able to configure the number with a var or a local ?

**Review by allamand — CHANGES_REQUESTED** [review]
Seems the repo directories has changed since you create this PR

File: examples/backend-service/main.tf (hunk 2 above)
Comment (bryantbiggs) [reply examples/backend-service/main.tf:83]: - module: A container for multiple resources that are used together to create lightweight abstractions. A module accepts inputs to configure and influence the resulting resources, and provides zero or more outputs. Modules are generally sourced from the Terraform registry but can also be sourced from a local directory, from GitHub, or a private registry.

- blueprint/example: An example implementation of one or more modules and/or one one or more resource objects that demonstrates an outcome. A blueprint/example is not intended to be used in-place nor is it intended to be used as a source like that found within a module. To use a blueprint/example, users can copy+paste the relevant code into their own project, or simply use a reference for how to structure their own code/project

**Comment by bryantbiggs** [comment]
@nvpnathan :shipit:

**Pull request merged**